src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Upsert/UpsertResourceHandler.cs

```diff
@@ -53,30 +53,26 @@ public async Task<UpsertResourceResponse> Handle(UpsertResourceRequest request,
 
             Resource resource = request.Resource.ToPoco<Resource>();
 
-            if (await ConformanceProvider.Value.RequireETag(resource.TypeName, cancellationToken) && request.WeakETag == null)
-            {
-                throw new PreconditionFailedException(string.Format(Core.Resources.IfMatchHeaderRequiredForResource, resource.TypeName));
-            }
-
```

Comment:
This check ensures the version is specified in the `if-match` header when the versioning policy is set to `versioned-update`.

In this PR, the check is moved to data store layer where we know if the upsert operation is an update or a create, allowing us to only throw an error if we are updating.